ui/src/app/pages/analysis/notebook-resource-card.tsx

```diff
@@ -0,0 +1,187 @@
+import {withErrorModal, WithErrorModalProps} from 'app/components/with-error-modal';
+import {withSpinnerOverlay, WithSpinnerOverlayProps} from 'app/components/with-spinner-overlay';
+import {CopyNotebookModal} from 'app/pages/analysis/copy-notebook-modal';
+import {workspacesApi} from 'app/services/swagger-fetch-clients';
+import colors from 'app/styles/colors';
+import {formatRecentResourceDisplayDate} from 'app/utils';
+import {navigateByUrl} from 'app/utils/navigation';
+import {RecentResource} from 'generated/fetch';
+import * as fp from 'lodash';
+import * as React from 'react';
+
+interface Props extends WithConfirmDeleteModalProps, WithErrorModalProps, WithSpinnerOverlayProps {
+  resourceCard: RecentResource;
+  onUpdate: Function;
+}
+
+interface State {
+  showCopyNotebookModal: boolean;
+  showRenameModal: boolean;
+}
+
+export const NotebookResourceCard = fp.flow(
+  withErrorModal(),
+  withConfirmDeleteModal(),
+  withSpinnerOverlay(),
+)(class extends React.Component<Props, State> {
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      showCopyNotebookModal: false,
+      showRenameModal: false
+    };
+  }
+
+  get resourceType(): string {
+    return 'Notebook';
+  }
+
+  get displayName(): string {
+    return this.props.resourceCard.notebook.name.replace(/\.ipynb$/, '');
+  }
+
+  get readOnly(): boolean {
+    return this.props.resourceCard.permission === 'READER';
+  }
+
+  get writePermission(): boolean {
+    return this.props.resourceCard.permission === 'OWNER'
+      || this.props.resourceCard.permission === 'WRITER';
+  }
+
+  get actions(): Action[] {
+    return [
+      {
+        icon: 'pencil',
+        displayName: 'Rename',
+        onClick: () => {
+          this.setState({showRenameModal: true});
+        },
+      },
+      {
+        icon: 'copy',
+        displayName: 'Duplicate',
+        onClick: () => this.duplicateNotebook(),
+      },
+      {
+        icon: 'copy',
+        displayName: 'Copy to another Workspace',
+        onClick: () => this.setState({showCopyNotebookModal: true}),
+      },
+      {
+        icon: 'trash',
+        displayName: 'Delete',
+        onClick: () => {
+          this.props.showConfirmDeleteModal(this.displayName,
+            this.resourceType, () => this.deleteNotebook());
+        },
+      },
+      {
+        icon: 'grid-view',
+        displayName: 'Open in Jupyter Lab',
+        onClick: () => navigateByUrl(this.resourceUrl(true)),
+      }];
+  }
+
+  fullNotebookName(name) {
+    return !name || /^.+\.ipynb$/.test(name) ? name : `${name}.ipynb`;
+  }
+
+  resourceUrl(jupyterLab = false): string {
+    const {workspaceNamespace, workspaceFirecloudName, notebook} =
+      this.props.resourceCard;
+    const workspacePrefix = `/workspaces/${workspaceNamespace}/${workspaceFirecloudName}`;
+
+    const queryParams = new URLSearchParams([
```

Comment:
`queryParams` are no longer used, and can be removed here.

This also means that...
* the `jupyterLab` function param can be removed
* the notebook resource card Action "Open in Jupyter Lab" can be removed, and
* the feature flag `enableJupyterLab` can be removed

... but it might be a good idea to check with Product on this.

At a minimum, disable the "Open in Jupyter Lab" Action if the feature flag `enableJupyterLab` is false.

Reply:
we decided to just remove it altogether since it's currently unsupported